### ISSUE TITLE
kubelet-NodeConformance test has continually failing test

### ISSUE THREAD

**Issue by mmerkes** [kubernetes/kubernetes#97130]
**Which jobs are failing**:

`kubelet-NodeConformance`

**Which test(s) are failing**:

`Pods should run through the lifecycle of Pods and PodStatus`

**Since when has it been failing**:

11/26/2020

**Testgrid link**:

https://testgrid.k8s.io/sig-node-critical#kubelet-NodeConformance

**Reason for failure**:

[The test was recently added to NodeConformance](https://github.com/kubernetes/kubernetes/pull/96485/files).

**Anything else we need to know**:

Commits:

* 739a698c0
* 5ed4b76a0

**Comment by mmerkes** [comment]
In the PR, [they reference the failing test as having flaked](https://github.com/kubernetes/kubernetes/pull/96485#issuecomment-732335344), and it's been proposed to [increase the timeout](https://github.com/kubernetes/kubernetes/pull/96691) to give the pods more time to get up and running. The `node-kubelet-conformance` job is run with only a single node, so it's possible that the reduced capacity impacts this.

That being said, we realized that this test wasn't actually supposed to run. [It's only tagged with [Conformance], not [NodeConformance]](https://github.com/ii/kubernetes/blob/16480d8ec028ba762edb84189b80f0dc6b8d93bd/test/conformance/testdata/conformance.yaml#L273), so it never should have been picked up in the first place.

Looking the logs, you can see that we set `--focus="\[NodeConformance\]"` when calling `run_remote`:

```
Running: go run /go/src/k8s.io/kubernetes/test/e2e_node/runner/remote/run_remote.go --cleanup --logtostderr --vmodule=*=4 --ssh-env=gce --results-dir=/workspace/_artifacts --project=k8s-jkns-gke-ubuntu-updown --zone=us-west1-b --ssh-user=prow --ssh-key=/workspace/.ssh/google_compute_engine --ginkgo-flags=--nodes=1 --focus="\[NodeConformance\]" --skip="\[Flaky\]" --test_args=--feature-gates=DynamicKubeletConfig=true --kubelet-flags="--cgroups-per-qos=true --cgroup-root=/" --test-timeout=1h5m0s --image-config-file=/workspace/test-infra/jobs/e2e_node/image-config.yaml --test-suite=conformance
```

However, if you look further, you can see that the ginkgo flags eventually change to `--ginkgo.focus=\\[Conformance\\]`:

```
Starting server "services" with command "/usr/local/bin/e2e_node.test --run-services-mode --bearer-token=EUadI1lYyJBdjfZn --test.timeout=24h0m0s --ginkgo.seed=1607548651 --ginkgo.focus=\\[Conformance\\] --ginkgo.skip=\\[Flaky\\]|\\[Serial\\] --ginkgo.flakeAttempts=2 --ginkgo.parallel.node=1 --ginkgo.parallel.total=8 --ginkgo.parallel.streamhost=http://127.0.0.1:37247 --ginkgo.parallel.synchost=http://127.0.0.1:37247 --ginkgo.slowSpecThreshold=5.00000 --conformance=true --prepull-images=false --report-dir=/var/result --system-spec-name= --system-spec-file= --extra-envs= --bearer-token=EUadI1lYyJBdjfZn"
```

When running the node conformance tests, [you can see that we drop the ginkgo args, see _ in args, and the "--ginkgo.focus=\\[Conformance\\]" flag gets dropped.](https://github.com/kubernetes/kubernetes/blob/7df379d75199bb28e0d9245fd844acc3499c7720/test/e2e_node/remote/node_conformance.go#L265) This chain all starts from setting the `--test-suite=conformance` flag, which [is only used by this job.](https://cs.k8s.io/?q=--test-suite%3Dconformance&i=nope&files=&repos=) As a result of setting this flag, we end up ignoring the ginkgo flags passed it and run the `[Conformance]` tests instead.

We have two options for fixing this.

1. Put this in line with the other tests, drop the `--test-suite=conformance` flag and rely on the `[NodeConformance]` tag to make it "conformance-y"
2. Track down all of the places that need to be changed to respect the ginkgo args passed in and not run the tests that only have the `[Conformance]` tag.

The SIG-Node CI subgroup has actually been talking about changing the node-kubelet-conformance tests for a while - [read more details here](https://github.com/kubernetes/test-infra/issues/18973). It's very similar to the `node-kubelet-master` tests, and the primary difference is that setting the `--test-suite=conformance` tag results in the tests getting run in Docker. Given the community's movement away from Docker and the potential changes required to make the `--test-suite=conformance` flag do the right thing without impacting any other areas, I think that we should go with option #1 above. It's simpler, will make this job more maintainable and if we're going to put effort into the current direction, we should probably revisit how it works in general.

I'm going to post a PR and we can discuss further here or there.

**Comment by ehashman** [comment]
@mmerkes what's the status of this issue? looks like the test suite is still failing to run

**Comment by pacoxu** [comment]
https://github.com/kubernetes/test-infra/issues/18973 seems related.
- https://github.com/kubernetes/kubernetes/issues/95877#issuecomment-721503322
#96128 fixed #95877 node-kubelet-conformance suite:  tests are failing in the BeforeSuite.

**Comment by ehashman** [comment]
/close

Fixed in https://github.com/kubernetes/test-infra/issues/22250 -- the tab/test job has been removed.
